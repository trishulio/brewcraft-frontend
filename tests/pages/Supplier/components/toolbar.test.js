import React from "react";
import { mount } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import { BrowserRouter } from "react-router-dom";
import Toolbar from "../../../../src/pages/Supplier/components/toolbar";

const initialState = {
    Supplier: {
        data: {
            id: "",
            name: "",
            address: {
                id: "",
                addressLine1: "",
                addressLine2: "",
                country: "",
                province: "",
                city: "",
                postalCode: "",
            },
            version: null,
        },
        initial: {
            id: "",
            name: "",
            address: {
                id: "",
                addressLine1: "",
                addressLine2: "",
                country: "",
                province: "",
                city: "",
                postalCode: "",
            },
            version: null,
        },
        invalidName: false,
        invalidAddressLine1: false,
        invalidAddressLIne2: false,
        invalidCountry: false,
        invalidProvince: false,
        invalidCity: false,
        inalidPostalCode: false,
        loading: true,
        error: true,
    },
};
const middlewares = [];

const mockStore = configureStore(middlewares);

const store = mockStore(initialState);
const mockDispatch = jest.fn();
const mockGoBack = jest.fn();
const mockPush = jest.fn();

const mockHistory = {
    replace: jest.fn(),
    goBack: mockGoBack,
    push: mockPush,
};

jest.mock("react-redux", () => ({
    ...jest.requireActual("react-redux"),
    useDispatch: () => mockDispatch,
}));

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useParams: jest.fn().mockReturnValue({
        id: "",
    }),
    useHistory: () =>
        mockHistory.mockReturnValue({
            goBack: () => mockGoBack,
        }),
}));

jest.mock("react-router", () => ({
    ...jest.requireActual("react-router"),
    useHistory: () => mockHistory,
}));

// configure({ adapter: new Adapter() });

describe("Supplier -> Components -> <Toolbar>", () => {
    describe("render()", () => {
        test("renders the toolbar component", () => {
            const toolBarComponent = mount(
                <Provider store={store}>
                    <BrowserRouter>
                        <Toolbar
                            editable={false}
                            changed={false}
                            onSave={jest.fn()}
                            onDelete={jest.fn()}
                        />
                    </BrowserRouter>
                </Provider>
            );
            expect(shallowToJson(toolBarComponent)).toMatchSnapshot();
        });
        test("History.goback button should be called on cancel", () => {
            const toolBarComponent = mount(
                <Provider store={store}>
                    <BrowserRouter>
                        <Toolbar editable={true} />
                    </BrowserRouter>
                </Provider>
            );
            toolBarComponent.find("Button").at(1).simulate("click");
            expect(mockGoBack).toHaveBeenCalled();
            expect(shallowToJson(toolBarComponent)).toMatchSnapshot();
        });
        test("History.push button should be called on edit", () => {
            initialState.Supplier.data.id = 1;
            const store = mockStore(initialState);
            const toolBarComponent = mount(
                <Provider store={store}>
                    <BrowserRouter>
                        <Toolbar editable={false} />
                    </BrowserRouter>
                </Provider>
            );
            toolBarComponent
                .findWhere((node) => {
                    return (
                        node.type() === "button" &&
                        node.text().includes("Cancel")
                    );
                })
                .at(0)
                .simulate("click");
            expect(mockGoBack).toHaveBeenCalled();
            expect(shallowToJson(toolBarComponent)).toMatchSnapshot();
        });

        test("History.push button should be called on Suppliers", () => {
            initialState.Supplier.data.id = 1;
            const store = mockStore(initialState);
            const toolBarComponent = mount(
                <Provider store={store}>
                    <BrowserRouter>
                        <Toolbar editable={false} />
                    </BrowserRouter>
                </Provider>
            );
            toolBarComponent
                .findWhere((node) => {
                    return (
                        node.type() === "button" &&
                        node.text().includes("Suppliers")
                    );
                })
                .at(0)
                .simulate("click");
            expect(mockGoBack).toHaveBeenCalled();
            expect(mockPush).toHaveBeenCalledWith("/suppliers");
            expect(shallowToJson(toolBarComponent)).toMatchSnapshot();
        });
        test("History.push button should be called on new Supplier", () => {
            initialState.Supplier.data.id = 1;
            const store = mockStore(initialState);
            const toolBarComponent = mount(
                <Provider store={store}>
                    <BrowserRouter>
                        <Toolbar editable={false} />
                    </BrowserRouter>
                </Provider>
            );
            toolBarComponent
                .findWhere((node) => {
                    return (
                        node.type() === "button" &&
                        node.text().includes("New Supplier")
                    );
                })
                .at(0)
                .simulate("click");
            expect(mockGoBack).toHaveBeenCalled();
            expect(mockPush).toHaveBeenCalledWith({
                pathname: "/suppliers/new",
                search: "?edit=true",
            });
            expect(shallowToJson(toolBarComponent)).toMatchSnapshot();
        });
    });
});
