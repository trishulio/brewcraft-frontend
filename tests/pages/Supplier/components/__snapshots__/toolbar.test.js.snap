// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Supplier -> Components -> <Toolbar> render() History.goback button should be called on cancel 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Toolbar
        editable={true}
      >
        <Button
          className="waves-effect mr-2 mb-3"
          color="primary"
          disabled={true}
          hidden={false}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-primary btn-sm disabled"
            disabled={true}
            hidden={false}
            onClick={[Function]}
            type="button"
          >
            Save
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          hidden={false}
          onClick={[Function]}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-secondary btn-sm"
            hidden={false}
            onClick={[Function]}
            type="button"
          >
            Cancel
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          disabled={true}
          hidden={true}
          onClick={[Function]}
          outline={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-outline-secondary btn-sm disabled"
            disabled={true}
            hidden={true}
            onClick={[Function]}
            type="button"
          >
            Edit
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="danger"
          hidden={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-danger btn-sm"
            hidden={true}
            onClick={[Function]}
            type="button"
          >
            Delete Supplier
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          hidden={true}
          onClick={[Function]}
          outline={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-outline-secondary btn-sm"
            hidden={true}
            onClick={[Function]}
            type="button"
          >
            Suppliers
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          hidden={true}
          onClick={[Function]}
          outline={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-outline-secondary btn-sm"
            hidden={true}
            onClick={[Function]}
            type="button"
          >
            New Supplier
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          hidden={true}
          onClick={[Function]}
          outline={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-outline-secondary btn-sm"
            hidden={true}
            onClick={[Function]}
            type="button"
          >
            New Contact
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          hidden={true}
          onClick={[Function]}
          outline={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-outline-secondary btn-sm"
            hidden={true}
            onClick={[Function]}
            type="button"
          >
            Contacts
          </button>
        </Button>
      </Toolbar>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`Supplier -> Components -> <Toolbar> render() History.push button should be called on Contacts 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Toolbar
        editable={false}
      >
        <Button
          className="waves-effect mr-2 mb-3"
          color="primary"
          disabled={true}
          hidden={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-primary btn-sm disabled"
            disabled={true}
            hidden={true}
            onClick={[Function]}
            type="button"
          >
            Save
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          hidden={true}
          onClick={[Function]}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-secondary btn-sm"
            hidden={true}
            onClick={[Function]}
            type="button"
          >
            Cancel
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          disabled={false}
          hidden={false}
          onClick={[Function]}
          outline={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-outline-secondary btn-sm"
            disabled={false}
            hidden={false}
            onClick={[Function]}
            type="button"
          >
            Edit
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="danger"
          hidden={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-danger btn-sm"
            hidden={true}
            onClick={[Function]}
            type="button"
          >
            Delete Supplier
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          hidden={false}
          onClick={[Function]}
          outline={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-outline-secondary btn-sm"
            hidden={false}
            onClick={[Function]}
            type="button"
          >
            Suppliers
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          hidden={false}
          onClick={[Function]}
          outline={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-outline-secondary btn-sm"
            hidden={false}
            onClick={[Function]}
            type="button"
          >
            New Supplier
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          hidden={false}
          onClick={[Function]}
          outline={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-outline-secondary btn-sm"
            hidden={false}
            onClick={[Function]}
            type="button"
          >
            New Contact
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          hidden={false}
          onClick={[Function]}
          outline={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-outline-secondary btn-sm"
            hidden={false}
            onClick={[Function]}
            type="button"
          >
            Contacts
          </button>
        </Button>
      </Toolbar>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`Supplier -> Components -> <Toolbar> render() History.push button should be called on New Contact 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Toolbar
        editable={false}
      >
        <Button
          className="waves-effect mr-2 mb-3"
          color="primary"
          disabled={true}
          hidden={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-primary btn-sm disabled"
            disabled={true}
            hidden={true}
            onClick={[Function]}
            type="button"
          >
            Save
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          hidden={true}
          onClick={[Function]}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-secondary btn-sm"
            hidden={true}
            onClick={[Function]}
            type="button"
          >
            Cancel
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          disabled={false}
          hidden={false}
          onClick={[Function]}
          outline={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-outline-secondary btn-sm"
            disabled={false}
            hidden={false}
            onClick={[Function]}
            type="button"
          >
            Edit
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="danger"
          hidden={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-danger btn-sm"
            hidden={true}
            onClick={[Function]}
            type="button"
          >
            Delete Supplier
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          hidden={false}
          onClick={[Function]}
          outline={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-outline-secondary btn-sm"
            hidden={false}
            onClick={[Function]}
            type="button"
          >
            Suppliers
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          hidden={false}
          onClick={[Function]}
          outline={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-outline-secondary btn-sm"
            hidden={false}
            onClick={[Function]}
            type="button"
          >
            New Supplier
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          hidden={false}
          onClick={[Function]}
          outline={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-outline-secondary btn-sm"
            hidden={false}
            onClick={[Function]}
            type="button"
          >
            New Contact
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          hidden={false}
          onClick={[Function]}
          outline={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-outline-secondary btn-sm"
            hidden={false}
            onClick={[Function]}
            type="button"
          >
            Contacts
          </button>
        </Button>
      </Toolbar>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`Supplier -> Components -> <Toolbar> render() History.push button should be called on Suppliers 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Toolbar
        editable={false}
      >
        <Button
          className="waves-effect mr-2 mb-3"
          color="primary"
          disabled={true}
          hidden={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-primary btn-sm disabled"
            disabled={true}
            hidden={true}
            onClick={[Function]}
            type="button"
          >
            Save
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          hidden={true}
          onClick={[Function]}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-secondary btn-sm"
            hidden={true}
            onClick={[Function]}
            type="button"
          >
            Cancel
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          disabled={false}
          hidden={false}
          onClick={[Function]}
          outline={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-outline-secondary btn-sm"
            disabled={false}
            hidden={false}
            onClick={[Function]}
            type="button"
          >
            Edit
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="danger"
          hidden={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-danger btn-sm"
            hidden={true}
            onClick={[Function]}
            type="button"
          >
            Delete Supplier
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          hidden={false}
          onClick={[Function]}
          outline={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-outline-secondary btn-sm"
            hidden={false}
            onClick={[Function]}
            type="button"
          >
            Suppliers
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          hidden={false}
          onClick={[Function]}
          outline={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-outline-secondary btn-sm"
            hidden={false}
            onClick={[Function]}
            type="button"
          >
            New Supplier
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          hidden={false}
          onClick={[Function]}
          outline={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-outline-secondary btn-sm"
            hidden={false}
            onClick={[Function]}
            type="button"
          >
            New Contact
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          hidden={false}
          onClick={[Function]}
          outline={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-outline-secondary btn-sm"
            hidden={false}
            onClick={[Function]}
            type="button"
          >
            Contacts
          </button>
        </Button>
      </Toolbar>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`Supplier -> Components -> <Toolbar> render() History.push button should be called on edit 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Toolbar
        editable={false}
      >
        <Button
          className="waves-effect mr-2 mb-3"
          color="primary"
          disabled={true}
          hidden={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-primary btn-sm disabled"
            disabled={true}
            hidden={true}
            onClick={[Function]}
            type="button"
          >
            Save
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          hidden={true}
          onClick={[Function]}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-secondary btn-sm"
            hidden={true}
            onClick={[Function]}
            type="button"
          >
            Cancel
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          disabled={false}
          hidden={false}
          onClick={[Function]}
          outline={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-outline-secondary btn-sm"
            disabled={false}
            hidden={false}
            onClick={[Function]}
            type="button"
          >
            Edit
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="danger"
          hidden={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-danger btn-sm"
            hidden={true}
            onClick={[Function]}
            type="button"
          >
            Delete Supplier
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          hidden={false}
          onClick={[Function]}
          outline={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-outline-secondary btn-sm"
            hidden={false}
            onClick={[Function]}
            type="button"
          >
            Suppliers
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          hidden={false}
          onClick={[Function]}
          outline={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-outline-secondary btn-sm"
            hidden={false}
            onClick={[Function]}
            type="button"
          >
            New Supplier
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          hidden={false}
          onClick={[Function]}
          outline={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-outline-secondary btn-sm"
            hidden={false}
            onClick={[Function]}
            type="button"
          >
            New Contact
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          hidden={false}
          onClick={[Function]}
          outline={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-outline-secondary btn-sm"
            hidden={false}
            onClick={[Function]}
            type="button"
          >
            Contacts
          </button>
        </Button>
      </Toolbar>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`Supplier -> Components -> <Toolbar> render() History.push button should be called on new Supplier 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Toolbar
        editable={false}
      >
        <Button
          className="waves-effect mr-2 mb-3"
          color="primary"
          disabled={true}
          hidden={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-primary btn-sm disabled"
            disabled={true}
            hidden={true}
            onClick={[Function]}
            type="button"
          >
            Save
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          hidden={true}
          onClick={[Function]}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-secondary btn-sm"
            hidden={true}
            onClick={[Function]}
            type="button"
          >
            Cancel
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          disabled={false}
          hidden={false}
          onClick={[Function]}
          outline={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-outline-secondary btn-sm"
            disabled={false}
            hidden={false}
            onClick={[Function]}
            type="button"
          >
            Edit
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="danger"
          hidden={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-danger btn-sm"
            hidden={true}
            onClick={[Function]}
            type="button"
          >
            Delete Supplier
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          hidden={false}
          onClick={[Function]}
          outline={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-outline-secondary btn-sm"
            hidden={false}
            onClick={[Function]}
            type="button"
          >
            Suppliers
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          hidden={false}
          onClick={[Function]}
          outline={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-outline-secondary btn-sm"
            hidden={false}
            onClick={[Function]}
            type="button"
          >
            New Supplier
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          hidden={false}
          onClick={[Function]}
          outline={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-outline-secondary btn-sm"
            hidden={false}
            onClick={[Function]}
            type="button"
          >
            New Contact
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          hidden={false}
          onClick={[Function]}
          outline={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-outline-secondary btn-sm"
            hidden={false}
            onClick={[Function]}
            type="button"
          >
            Contacts
          </button>
        </Button>
      </Toolbar>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`Supplier -> Components -> <Toolbar> render() renders the toolbar component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Toolbar
        changed={false}
        editable={false}
        onDelete={[MockFunction]}
        onSave={[MockFunction]}
      >
        <Button
          className="waves-effect mr-2 mb-3"
          color="primary"
          disabled={true}
          hidden={true}
          onClick={[MockFunction]}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-primary btn-sm disabled"
            disabled={true}
            hidden={true}
            onClick={[Function]}
            type="button"
          >
            Save
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          hidden={true}
          onClick={[Function]}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-secondary btn-sm"
            hidden={true}
            onClick={[Function]}
            type="button"
          >
            Cancel
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          disabled={false}
          hidden={true}
          onClick={[Function]}
          outline={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-outline-secondary btn-sm"
            disabled={false}
            hidden={true}
            onClick={[Function]}
            type="button"
          >
            Edit
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="danger"
          disabled={false}
          hidden={true}
          onClick={[MockFunction]}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-danger btn-sm"
            disabled={false}
            hidden={true}
            onClick={[Function]}
            type="button"
          >
            Delete Supplier
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          hidden={true}
          onClick={[Function]}
          outline={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-outline-secondary btn-sm"
            hidden={true}
            onClick={[Function]}
            type="button"
          >
            Suppliers
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          hidden={true}
          onClick={[Function]}
          outline={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-outline-secondary btn-sm"
            hidden={true}
            onClick={[Function]}
            type="button"
          >
            New Supplier
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          hidden={true}
          onClick={[Function]}
          outline={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-outline-secondary btn-sm"
            hidden={true}
            onClick={[Function]}
            type="button"
          >
            New Contact
          </button>
        </Button>
        <Button
          className="waves-effect mr-2 mb-3"
          color="secondary"
          hidden={true}
          onClick={[Function]}
          outline={true}
          size="sm"
          tag="button"
          type="button"
        >
          <button
            aria-label={null}
            className="waves-effect mr-2 mb-3 btn btn-outline-secondary btn-sm"
            hidden={true}
            onClick={[Function]}
            type="button"
          >
            Contacts
          </button>
        </Button>
      </Toolbar>
    </Router>
  </BrowserRouter>
</Provider>
`;
